streams/runners/storm/runtime/src/main/java/com/hortonworks/streamline/streams/runtime/storm/spout/AvroStreamsSnapshotDeserializer.java

```diff
@@ -65,7 +65,10 @@ private Object convertValue(Object deserializedObj) {
             List<Schema.Field> fields = indexedRecord.getSchema().getFields();
             ImmutableMap.Builder<String, Object> keyValues = ImmutableMap.builder();
             for (Schema.Field field : fields) {
-                keyValues.put(field.name(), convertValue(indexedRecord.get(field.pos())));
+                Object currentValue = convertValue(indexedRecord.get(field.pos()));
+                if (currentValue != null) {
```

Comment:
@harshach Agree that it is risky to have big changes now for this release. 

@priyank5485 @harshach @arunmahadevan Having Map abstraction exposed with StreamlineEvent is a leaky abstraction and we had discussion on that earlier. Ideally, internal implementation should not dictate the user API. Having Optional.empty() looks fine even though it adds burden on user to have this check instead of null. This would have been seamless if we have StreamlineEvent as Map<String, Optional> (with values always wrapped in Optional) instead of Map<String, Object>.